aws-iot-topicrule/src/main/java/com/amazonaws/iot/topicrule/BaseHandlerStd.java

```diff
@@ -0,0 +1,54 @@
+package com.amazonaws.iot.topicrule;
+
+import software.amazon.awssdk.services.iot.IotClient;
+import software.amazon.awssdk.services.iot.model.ListTagsForResourceResponse;
+import software.amazon.awssdk.services.iot.model.Tag;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.delay.Constant;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class BaseHandlerStd extends BaseHandler<CallbackContext> {
+    protected static final Constant DELAY_CONSTANT = Constant.of().timeout(Duration.ofMinutes(10L))
+            .delay(Duration.ofSeconds(30L)).build();
+
+    @Override
+    public final ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+
+        return handleRequest(
+                proxy,
+                request,
+                callbackContext != null ? callbackContext : new CallbackContext(),
+                proxy.newProxy(ClientBuilder::getClient),
+                logger);
+    }
+
+    protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<IotClient> proxyClient,
+            final Logger logger);
+
+    protected List<Tag> listTags(final ProxyClient<IotClient> proxyClient, final String arn) {
+        String nextToken = null;
+        List<Tag> listOfTags = new ArrayList<>();
+        do {
+            final ListTagsForResourceResponse response = proxyClient.injectCredentialsAndInvokeV2(Translator.translateToListTagsRequest(arn, nextToken),
```

Comment:
can we use the streaming pagination api for list tags instead? 

The basic idea is described here : https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/pagination.html